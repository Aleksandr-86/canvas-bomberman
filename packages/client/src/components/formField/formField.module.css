.field {
    margin: 10px 0;
    padding: 0 10px;
}

.label {
    margin: 0 auto;
}

.input {
    max-width: 250px;
    background-color: transparent;
    border: 0;
    outline: none;
    text-align: center;
    padding-bottom: 5px;
    border-bottom: 5px solid black;
    font-size: 25px;
    line-height: 30px;
    font-family: 'Rubik Bubbles', fantasy;
}

.error {
    color: red;
    text-align: center;
    height: 20px;
    font-size: .75rem;
    margin-top: .25rem;
}
