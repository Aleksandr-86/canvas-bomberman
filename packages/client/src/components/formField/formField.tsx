import styles from './formField.module.css'

interface FormFieldProps {
  label?: string
  placeholder?: string
  name: string
  type: 'text' | 'password' | 'email' | 'tel'
  value: string
  error?: string
  onChange: { (event: React.ChangeEvent<HTMLInputElement>): void }
  onBlur?: { (event: React.FocusEvent<HTMLInputElement>): void }
  disabled?: boolean
  children?: any
}

export const FormField = (props: FormFieldProps) => {
  const {
    children,
    label,
    placeholder,
    name,
    value,
    error,
    onChange,
    onBlur,
    type,
    disabled,
  } = props

  return (
    <div className={styles.field}>
      <label className={styles.label}>{label}</label>

      <div>
        <input
          className={styles.input}
          placeholder={placeholder}
          name={name}
          type={type}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          disabled={disabled}
        />
        {children}
      </div>

      <p className={styles.error}>{disabled ? '' : error}</p>
    </div>
  )
}
