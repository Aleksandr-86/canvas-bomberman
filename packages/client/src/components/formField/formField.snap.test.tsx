import * as renderer from 'react-test-renderer'
import { FormField } from './formField'

const mockProps = {
  name: `mockName`,
  value: `mockValue`,
}

const onChangeMock = () => undefined

describe(`<FormField/> snapshot test cases`, () => {
  test(`<FormField/> should render <input> with type text (default)`, () => {
    const tree = renderer
      .create(
        <FormField {...mockProps} type={'text'} onChange={onChangeMock} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<FormField/> should render <input> with type password (type password)`, () => {
    const tree = renderer
      .create(
        <FormField {...mockProps} type={'password'} onChange={onChangeMock} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<FormField/> should render <input> with type email (type email)`, () => {
    const tree = renderer
      .create(
        <FormField {...mockProps} type={'email'} onChange={onChangeMock} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<FormField/> should render <input> with type tel (type tel)`, () => {
    const tree = renderer
      .create(<FormField {...mockProps} type={'tel'} onChange={onChangeMock} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<FormField/> should render <div className=field}> with childrenElement`, () => {
    const tree = renderer
      .create(
        <FormField {...mockProps} type={'tel'} onChange={onChangeMock}>
          {<div>childrenElement</div>}
        </FormField>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<FormField/> should render <div className=field}> with mockErrorText (disabled FALSE)`, () => {
    const tree = renderer
      .create(
        <FormField
          {...mockProps}
          type={'tel'}
          onChange={onChangeMock}
          disabled={false}
          error="mockErrorText"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
