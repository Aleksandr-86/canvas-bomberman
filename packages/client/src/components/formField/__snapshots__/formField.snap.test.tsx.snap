// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField/> snapshot test cases <FormField/> should render <div className=field}> with childrenElement 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      name="mockName"
      onChange={[Function]}
      type="tel"
      value="mockValue"
    />
    <div>
      childrenElement
    </div>
  </div>
  <p
    className="error"
  />
</div>
`;

exports[`<FormField/> snapshot test cases <FormField/> should render <div className=field}> with mockErrorText (disabled FALSE) 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      disabled={false}
      name="mockName"
      onChange={[Function]}
      type="tel"
      value="mockValue"
    />
  </div>
  <p
    className="error"
  >
    mockErrorText
  </p>
</div>
`;

exports[`<FormField/> snapshot test cases <FormField/> should render <input> with type email (type email) 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      name="mockName"
      onChange={[Function]}
      type="email"
      value="mockValue"
    />
  </div>
  <p
    className="error"
  />
</div>
`;

exports[`<FormField/> snapshot test cases <FormField/> should render <input> with type password (type password) 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      name="mockName"
      onChange={[Function]}
      type="password"
      value="mockValue"
    />
  </div>
  <p
    className="error"
  />
</div>
`;

exports[`<FormField/> snapshot test cases <FormField/> should render <input> with type tel (type tel) 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      name="mockName"
      onChange={[Function]}
      type="tel"
      value="mockValue"
    />
  </div>
  <p
    className="error"
  />
</div>
`;

exports[`<FormField/> snapshot test cases <FormField/> should render <input> with type text (default) 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div>
    <input
      className="input"
      name="mockName"
      onChange={[Function]}
      type="text"
      value="mockValue"
    />
  </div>
  <p
    className="error"
  />
</div>
`;
